aten/src/ATen/native/Normalization.cpp

```diff
@@ -199,6 +199,15 @@ std::tuple<Tensor,Tensor> batch_norm_cpu_update_stats_template(
   using accscalar_t = at::acc_type<scalar_t, false>;
 
   int64_t n_input = input.size(1);
+  TORCH_CHECK(
+      input.size(0) > 0 && n_input > 0 && input.numel() > 0,
+      "input.size(0), input.size(1) and input.numel() need to be greater than 0, got input.size(0) = ",
+      input.size(0),
+      ", input.size(1) = ",
+      n_input,
+      ", input.numel() = ",
+      input.numel()
+  );
```

Comment:
if size of any of the dimensions is 0, therefore checking only for `input.numel()` is sufficient. And along the same venue, why not just print a tensor shape?
```suggestion
  TORCH_CHECK(input.numel() != 0, "input must be a non-empty tensor, but got  ", input.sizes());
```

Reply:
@malfet Thanks for the confirmation that numel check is enough, was afraid missing some corner cases. Modified to  `TORCH_CHECK(input.numel() != 0, "input tensor must have at least one element, but got input_sizes = ", input.sizes());`

Did not use non-empty tensor since people may conflate this with torch.empty...
For example, input with torch.empty([1,1]) is OK. (It is actually not an "empty" tensor but people may not know due to the name).